lib/util/esWrapper.js

```diff
@@ -135,44 +137,36 @@ class ESWrapper {
         'elasticsearch_service_not_connected');
     }
 
-    const message = _.get(error, 'meta.body.error.reason') || '';
-
-    let kuzzleError = null;
+    const message = _.get(error, 'meta.body.error.reason', error.message);
 
     // Try to match a known elasticsearch error
     for (const mapping of errorMessagesMapping) {
       const matches = message.match(mapping.regex);
 
       if (matches) {
-        kuzzleError = errorsManager.getError(
+        return errorsManager.getError(
           'external',
           'elasticsearch',
           mapping.subcode,
           ...mapping.getPlaceholders(matches));
-
-        break;
       }
     }
 
-    switch (error.meta.statusCode) {
-      case 400:
-        kuzzleError = kuzzleError || this._handleBadRequestError(error, message);
-        break;
-
-      case 404:
-        kuzzleError = kuzzleError || this._handleNotFoundError(error, message);
-        break;
-
-      case 409:
-        kuzzleError = kuzzleError || this._handleConflictError(error, message);
-        break;
-
-      default:
-        kuzzleError = kuzzleError || this._handleUnknownError(error, message);
-        break;
+    // Try to match using error codes
+    if (error.meta) {
+      switch (error.meta.statusCode) {
+        case 400:
+          return this._handleBadRequestError(error, message);
+        case 404:
+          return this._handleNotFoundError(error, message);
+        case 409:
+          return this._handleConflictError(error, message);
+        default:
+          break;
+      }
     }
 
-    return kuzzleError;
+    return this._handleUnknownError(error, message);
```

Comment:
Why its not inside the default of your switch case above?

Reply:
Because we may not enter the switch if there is no error meta.